File: multi_thread/multi_thread.cpp
#include<iostream>
#include<thread>

void print_message(const std::string& message){
    std::cout << "Thread ID: " << std::this_thread::get_id() << " - Message: " << message << std::endl;
}


int main(){
    std::thread thread1(print_message, "Hello from thread 1");
    std::thread thread2(print_message, "Hello from thread 2");

    thread1.join();
    thread2.join();

    return 0;
}


// compile: g++  multi_thread.cpp -o multi_thread -pthread
// run: ./multi_thread

File: multi_thread/multi_thread.py
import threading

def print_message(message):
    print(f"Thread ID: {threading.current_thread().name} - message: {message}")
    
    
if __name__ == "__main__":
    thread1 = threading.Thread(target=print_message, args=("From thread 1",))
    thread2 = threading.Thread(target=print_message, args=("From thread 2",))
    
    thread1.start()
    thread2.start()
    
    thread1.join()
    thread2.join()